bundles/af-core/src/main/java/com/adobe/cq/forms/core/components/util/AbstractFormComponentImpl.java

```diff
@@ -155,12 +156,12 @@ public String getText() {
     @Override
     public String getName() {
         if (name == null) {
-            name = getDefaultName();
+            name = getNameOrSetDefault();
         }
         return name;
     }
 
-    protected String getDefaultName() {
+    protected String getNameOrSetDefault() {
```

Comment:
@AnurudraS ,do name the method appropriately, by the name it looks like this method will check if name is not null then return the current value of null otherwise will set the name and then return. Either change the name of move the null check from getName to here.
Why do you need a separate method, why can't you do this in the getName method only if name is null.

Reply:
> 
On removing the name the code is lossing the code coverage because we cannot write test case for name = null as the resource cannot be null or the Key name cannot be set to null. However the method is kept protected so it should not break or impact the other properties.